app/src/main/java/com/github/studydistractor/sdp/ui/LoginScreen.kt

```diff
@@ -1,23 +1,129 @@
 package com.github.studydistractor.sdp.ui
 
+import android.content.ContentValues
+import android.text.TextUtils
+import android.util.Log
+import android.widget.Toast
 import androidx.compose.foundation.layout.*
-import androidx.compose.material3.MaterialTheme
-import androidx.compose.material3.Text
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Email
+import androidx.compose.material.icons.filled.Lock
+import androidx.compose.material3.*
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.platform.testTag
-import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.unit.dp
+import com.github.studydistractor.sdp.login.LoginAuthInterface
 
+/**
+ * Author: Bluedrack and AdrienBouquet
+ */
+@OptIn(ExperimentalMaterial3Api::class)
 @Composable
-@Preview("Login Screen")
-fun LoginScreen() {
-    Screen {
-        Column() {
-            Text(
-                text = "Login screen.",
-                style = MaterialTheme.typography.titleLarge,
-                modifier = Modifier.testTag("login")
-            )
+fun LoginScreen(
+    onRegisterButtonClicked: () -> Unit,
+    onLoggedIn: () -> Unit,
+    loginAuth : LoginAuthInterface
+) {
+    val context = LocalContext.current
+    Column (
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(16.dp)
+            .testTag("login-screen__main-container"),
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        val email = remember { mutableStateOf(TextFieldValue("")) }
+        val password = remember { mutableStateOf(TextFieldValue("")) }
+        Text(
+            text = "Log In",
+            style = MaterialTheme.typography.titleLarge,
+            modifier = Modifier.padding(vertical = 32.dp)
+        )
+
+        OutlinedTextField(
+            value = email.value,
+            label = { Text(text = "Email") },
+            onValueChange = {
+                email.value = it
+            },
+            keyboardOptions = KeyboardOptions.Default.copy(
+                keyboardType = KeyboardType.Email
+            ),
+            leadingIcon = {
+                Icon(Icons.Filled.Email, contentDescription = null)
+            },
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(vertical = 8.dp)
+                .testTag("email")
+        )
+        OutlinedTextField(

```

Comment:
Need more description #65

Reply:
Ok, let's take care of it during refactoring.